### DISCUSSION TITLE
How to let initial request behaviour in cy.intercept?

### DISCUSSION THREAD

**Discussion by sunpietro** [cypress-io/cypress#16501] [Questions and Help] [answered]
I've overwritten the `cy.visit` command the following way:
```js
Cypress.Commands.overwrite('visit', (originalVisit, url, options) => {
    let originalVisitOptions = options;
    let shouldSkip = false;

    if (options && Object.prototype.hasOwnProperty.call(options, 'skip')) {
        const { skip, ...rest } = options;

        shouldSkip = skip;
        originalVisitOptions = rest;
    }

    cy.intercept('GET', 'some-url/**', (req) => {
        if (shouldSkip) {
            req.reply({
                statusCode: 503,
            });
        }
    });

    originalVisit(url, originalVisitOptions);
});
```
I'm using it to skip requesting some non-essential thrid-party libs to load on my website. I would like to revert its action when `shouldSkip` is set to `false` in some tests later.

I'm wondering how can I _revert_ the `req.reply` from if statement earlier?

**Comment by jennifer-shehane** [comment, marked as answer]
Assuming you are using the most recent version of Cypress, you should be able to define another `cy.intercept()` later that calls `req.reply()` or `req.continue` that will override the `cy.intercept()` defined previously. So, for the definition that overrides you can remove the statusCode and it should go through normally.

**Reply by sunpietro** [reply]
Thank you for your answer. I did it the following way (with `cy.reply()`) and it works:
```
    cy.intercept('GET', 'some-url/**', (req) => {
        if (shouldSkip) {
            return req.reply({
                statusCode: 503,
            });
        }

        return req.reply();
    });
```
